src/static/helpers/customApi.ts

```diff
@@ -134,11 +134,34 @@ export const callCustomEndpoint = async (args: {
     };
   }
 
+  const currentEndpointPath = (options?.customApiPathParameters?.endpointPath ||
+    clientConfig.parameters?.endpointPath) as string;
+  let newEndpointPath = currentEndpointPath;
+  const endpointPathSegments = {} as Record<string, string>;
+
+  // Normalize and template the endpointPath so each segment is encoded as a path param.
+  // Example:
+  //    currentEndpointPath: "action/categories/Special,Summer" ->
+  //    endpointPathParams: { endpointPathSegment0: "action", endpointPathSegment1: "categories", endpointPathSegment2: "Special,Summer" }
+  //    newEndpointPath: "{endpointPathSegment0}/{endpointPathSegment1}/{endpointPathSegment2}/"
+  // The TemplateURL will then encode the path parameters and construct the URL with the encoded path parameters
+  // The resulting endpointPath will be: "actions/categories/Special%2CSummer"
+  if (currentEndpointPath.indexOf('/') > 0) {
+    newEndpointPath = '';
+    currentEndpointPath.split('/').forEach((segment: string, index) => {
+      const key = `endpointPathSegment${index}`;
+      endpointPathSegments[key] = segment;
+      newEndpointPath += `{${key}}/`;
+    });
```

Comment:
Should we handle consecutive slashes like `'path//endpoint'` or leading slashes like `'/path/endpoint'` where it should normalize to `'path/endpoint'` and not `'path//endpoint'`?

```suggestion
  const segments = currentEndpointPath.split('/').filter(segment => segment !== '');
  
  segments.forEach((segment: string, index) => {
    const key = `endpointPathSegment${index}`;
    endpointPathSegments[key] = segment;
    newEndpointPath += `{${key}}/`;
  });
```